### PR TITLE
pyenv: new port

### PR DESCRIPTION
**Pull request by gleeblezoid**
#### Description

Adds pyenv 2.3.2, a Python version manager tool, to python ports.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.4 21F79 x86_64
Xcode 13.4.1 13F100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by reneeotten** [comment]
hi @gleeblezoid - thanks for the submission! There are a few details to polish-up but we can guide you through that part.

Before doing so though, my first question would be: what does this package accomplish that cannot be done using MacPorts' built-in "port select" mechanism?

**Comment by gleeblezoid** [comment]
> hi @gleeblezoid - thanks for the submission! There are a few details to polish-up but we can guide you through that part.
> 
> Before doing so though, my first question would be: what does this package accomplish that cannot be done using MacPorts' built-in "port select" mechanism?

Heya, thanks for picking up my PR so fast! 

The main things that I see being distinct are:

- version interception is automated with pyenv, you shim in a redirect to the correct python version automatically when you run your code/script/interactive shell (to my knowledge `port select` requires a conscious intervention by the user).
- project collaboration across platforms, pyenv makes collaborating on multiple Python projects across different versions _much_ easier in a more system agnostic way. If I'm a macOS user and half my team use Linux we can still standardise our Python dev environments via pyenv + pipenv to make that not matter (which is not a use-case supported by `port select` alone).

No worries if that doesn't hit the bar - I'd like pyenv to be a port but understand that the port list needs to be maintainable.

**Comment by reneeotten** [comment]
@gleeblezoid - I am not familiar enough with `pyenv` to see a real benefit here. Perhaps my example of `port select` wasn't the clearest. But... one can already install all the various Python versions side-by-side with MacPorts and just start them with `python-X.Y` without the need of any shims or anything (unless you want to be able to switch to things like "Anaconda" or even system-wide Python versions?). 

Regarding the CI failure, that should definitely pass and tells you there is something wrong with your Portfile. I will leave a few comments to improve/fix that.

File: python/pyenv/Portfile
Comment (reneeotten) [inline python/pyenv/Portfile:7]: this probably fits better in  "sysutils" as the primary category?

**Comment by gleeblezoid** [comment]
> @gleeblezoid - I am not familiar enough with `pyenv` to see a real benefit here. Perhaps my example of `port select` wasn't the clearest. But... one can already install all the various Python versions side-by-side with MacPorts and just start them with `python-X.Y` without the need of any shims or anything (unless you want to be able to switch to things like "Anaconda" or even system-wide Python versions?).

The automated aspect of pyenv is the main benefit over this to my mind, it will do things like automatically detect when a virtualenv should be used based on working directory and activate it, and also direct code to the correct interpreter without the user needing to initiate that switch using a version specific python command.

MacPorts allowing safe side-by-side operation of multiple Python versions is definitely an overlapping benefit but not the sole (or primary) utility one gets from pyenv to my mind.

Also thank you for the pointers on my Portfile, much appreciated. 👍🏾

File: python/pyenv/Portfile
Comment (reneeotten) [inline python/pyenv/Portfile:4]: you should be using the [`github` PG](https://guide.macports.org/index.html#reference.portgroup.github)

File: python/pyenv/Portfile
Comment (reneeotten) [inline python/pyenv/Portfile:6]: use the `github.setup` syntax instead of specifying `name` and `version` separately

File: python/pyenv/Portfile
Comment (reneeotten) [inline python/pyenv/Portfile:14]: `homepage` and `master_sites` will be set automagically from the `github.setup` line

File: python/pyenv/Portfile
Comment (reneeotten) [reply python/pyenv/Portfile:6]: and use `github.tarball_from archive'

File: python/pyenv/Portfile
Comment (reneeotten) [inline python/pyenv/Portfile:15]: make sure to remove you local distfile and recalculate the checksums: `sudo port clean --all pyenv ; sudo port -dv checksums pyenv`

File: python/pyenv/Portfile
Comment (reneeotten) [inline python/pyenv/Portfile:18]: where do these dependencies come into play? For just being a "shim" for different Python versions this list seemsa little weird to me...

**Review by reneeotten — CHANGES_REQUESTED** [review]
a few comments after an initial quick read - haven't tested anything myself

File: python/pyenv/Portfile
Comment (gleeblezoid) [reply python/pyenv/Portfile:18]: They're based around providing dependencies for subsequent installs of Python - not essential but they do mean you avoid potential "weird" issues when pyenv compiles the Python version you're bringing in https://github.com/pyenv/pyenv/wiki#suggested-build-environment

File: python/pyenv/Portfile
Comment (gleeblezoid) [reply python/pyenv/Portfile:7]: Cool - I picked devel on the grounds I've primarily used it as a dev tool but sysutils definitely makes sense in a more neutral context.

File: python/pyenv/Portfile
Comment (gleeblezoid) [reply python/pyenv/Portfile:4]: _This_ makes everything make more sense, thank you.

**Comment by gleeblezoid** [comment]
Awesome - thank you so much for the guidance here @reneeotten - I'll go back and fix this stuff up. 

Once I've got all those changes up and all's good I'll squash everything per the guidelines to not clutter up commit history.

File: python/pyenv/Portfile
Comment (reneeotten) [reply python/pyenv/Portfile:18]: ah, now I remember.... that's the part I personally don't like about this package.... We should use MacPorts to install Python versions not some third-party tool ;)

Comment (gleeblezoid) [reply python/pyenv/Portfile:18]: Sadly people on dev teams insist on using different OS's and package managers, they have not seen the light. 😭 😂

Comment (gleeblezoid) [reply python/pyenv/Portfile:18]: That said, I am running a steady campaign to get my current team to all move to MacPorts…

**Comment by gleeblezoid** [comment]
Looks like this got further but then failed in the ./configure step - I'll investigate to get the install working locally and then commit + re-request review at that point.

**Comment by reneeotten** [comment]
Yeah, there is no "configure" and the Makefile seems a bit weird as they only do testing.

I now see that you removed the "tried a local install" in the PR checklist; none of these steps are  actually optional 😉

**Comment by gleeblezoid** [comment]
> Yeah, there is no "configure" and the Makefile seems a bit weird as they only do testing.
> 
> I now see that you removed the "tried a local install" in the PR checklist; none of these steps are actually optional 😉

That was indeed me being a potato (I assumed that chicken/egg issue I mentioned), that led me to remove them based on the `<!-- (delete not applicable items) -->` flag. At least I was committed in my path to silliness :lolsob:

**Comment by gleeblezoid** [comment]
Serious thanks for the unending patience and pointers here by the way, chances are I'll want to contribute in the future and I really appreciate you setting me up for success. 🙇🏾

**Comment by gleeblezoid** [comment]
I think the changes need to be approved before this can be merged fwiw - but let me know if anything is needed from me and if I'm good to squash the commits.

**Comment by mascguy** [comment]
> I think the changes need to be approved before this can be merged fwiw - but let me know if anything is needed from me and if I'm good to squash the commits.

Yep, there are some minor changes that will be needed, and I'm reviewing presently. Will provide more detail shortly.

**Comment by mascguy** [comment]
> Yep, there are some minor changes that will be needed, and I'm reviewing presently. Will provide more detail shortly.

Pushed an initial round of changes, with two key things:
* Minor formatting/reorganization, per established precedent and best practices;
* Use of existing phases `configure` and `build`, rather than hard-coding that logic in phase `destroot`.

**Comment by mascguy** [comment]
While CI looks good, I don't consider this complete just yet: When CCache is in the mix, the configure script fails with the following:

```
:info:configure aborted: compiler not found: ccache /usr/bin/clang
```

Looks reasonably easy to fix, though.

**Comment by mascguy** [comment]
We also need to copy the non-bin related files - such as readmes, etc - to the standard locations. (`${prefix}/share/...`.)

**Comment by mascguy** [comment]
> While CI looks good, I don't consider this complete just yet: When CCache is in the mix, the configure script fails with the following:

Fixed by simply disabling CCache. Doesn't benefit us here anyway, so no need to patch configure script.

**Comment by mascguy** [comment]
> We also need to copy the non-bin related files - such as readmes, etc - to the standard locations. (`${prefix}/share/...`.)

Currently reviewing upstream's logic in their various related projects, particularly: https://github.com/pyenv/pyenv-installer

**Comment by gleeblezoid** [comment]
Thanks @mascguy and sorry for not replying this past week! I'll see if I can figure out how to add those extra bits (I think there's a bunch of existing ports I can use for inspiration).

**Comment by gleeblezoid** [comment]
I've done a commit squash in case these are all the changes needed - but if not then no worries at all - just figure it's neater and might save pain later if a re-squash is needed.

**Comment by mascguy** [comment]
> I've done a commit squash in case these are all the changes needed - but if not then no worries at all - just figure it's neater and might save pain later if a re-squash is needed.

Let me know if you'd like to make any further changes.

There's no rush, and take as much time as you need. But if you're done, I'm happy to merge.

**Comment by gleeblezoid** [comment]
Awesome - thank you! Sorry about the really late reply, didn't spot the comment and got distracted by shiny/life things. Happy to merge and thanks so much for the help @mascguy !

**Comment by thecesrom** [comment]
Is it possible to side-load this Portfile and test it?

I switched from Homebrew and I do miss having the ability to install `pyenv`, as having to install `py-pip`, `py-setuptools` and others was quite a surprise; but I'm sure there's a good reason for that.

The reason why I'm saying this is because I did have a problem building Python 3.10.8 where it warned be about missing `lzma`, even though I did install the listed ports (`pkgconfig` `openssl` `xz` `gdbm`) from Python's [official guide](https://devguide.python.org/getting-started/setup-building/index.html#macos-and-os-x).

```sh
$ sudo port install pkgconfig openssl xz gdbm
$ pyenv install 3.10.8
Downloading openssl-1.1.1o.tar.gz...
-> https://www.openssl.org/source/openssl-1.1.1o.tar.gz
Installing openssl-1.1.1o...
Installed openssl-1.1.1o to /Users/thecesrom/.pyenv/versions/3.10.8

Downloading readline-8.1.tar.gz...
-> https://ftpmirror.gnu.org/readline/readline-8.1.tar.gz
Installing readline-8.1...
Installed readline-8.1 to /Users/thecesrom/.pyenv/versions/3.10.8

Downloading Python-3.10.8.tar.xz...
-> https://www.python.org/ftp/python/3.10.8/Python-3.10.8.tar.xz
Installing Python-3.10.8...
python-build: use zlib from xcode sdk
WARNING: The Python lzma extension was not compiled. Missing the lzma lib?
Installed Python-3.10.8 to /Users/thecesrom/.pyenv/versions/3.10.8
```

To overcome this, as suggested by the Setup and Building Python guide, and `pyenv`'s [Common build problems](https://github.com/pyenv/pyenv/wiki/Common-build-problems) one should set `CPPFLAGS` and `LDFLAGS` like this:

```sh
CPPFLAGS="-I/opt/local/include" LDFLAGS="-L/opt/local/lib" pyenv install 3.10.8
Downloading openssl-1.1.1o.tar.gz...
-> https://www.openssl.org/source/openssl-1.1.1o.tar.gz
Installing openssl-1.1.1o...
Installed openssl-1.1.1o to /Users/thecesrom/.pyenv/versions/3.10.8

Downloading readline-8.1.tar.gz...
-> https://ftpmirror.gnu.org/readline/readline-8.1.tar.gz
Installing readline-8.1...
Installed readline-8.1 to /Users/thecesrom/.pyenv/versions/3.10.8

Downloading Python-3.10.8.tar.xz...
-> https://www.python.org/ftp/python/3.10.8/Python-3.10.8.tar.xz
Installing Python-3.10.8...
python-build: use zlib from xcode sdk
Installed Python-3.10.8 to /Users/thecesrom/.pyenv/versions/3.10.8
```

Also, `pyenv` uses `openssl` and `readline` from Homebrew if present, something I am not seeing here.

So my question is.

How were you (@gleeblezoid) able to install a Python version with `pyenv` without setting those flags?

Thanks

**Comment by gleeblezoid** [comment]
> Is it possible to side-load this Portfile and test it?
> 
> I switched from Homebrew and I do miss having the ability to install `pyenv`, as having to install `py-pip`, `py-setuptools` and others was quite a surprise; but I'm sure there's a good reason for that.
> 
> The reason why I'm saying this is because I did have a problem building Python 3.10.8 where it warned be about missing `lzma`, even though I did install the listed ports (`pkgconfig` `openssl` `xz` `gdbm`) from Python's [official guide](https://devguide.python.org/getting-started/setup-building/index.html#macos-and-os-x).
> 
> ```sh
> $ sudo port install pkgconfig openssl xz gdbm
> $ pyenv install 3.10.8
> Downloading openssl-1.1.1o.tar.gz...
> -> https://www.openssl.org/source/openssl-1.1.1o.tar.gz
> Installing openssl-1.1.1o...
> Installed openssl-1.1.1o to /Users/thecesrom/.pyenv/versions/3.10.8
> 
> Downloading readline-8.1.tar.gz...
> -> https://ftpmirror.gnu.org/readline/readline-8.1.tar.gz
> Installing readline-8.1...
> Installed readline-8.1 to /Users/thecesrom/.pyenv/versions/3.10.8
> 
> Downloading Python-3.10.8.tar.xz...
> -> https://www.python.org/ftp/python/3.10.8/Python-3.10.8.tar.xz
> Installing Python-3.10.8...
> python-build: use zlib from xcode sdk
> WARNING: The Python lzma extension was not compiled. Missing the lzma lib?
> Installed Python-3.10.8 to /Users/thecesrom/.pyenv/versions/3.10.8
> ```
> 
> To overcome this, as suggested by the Setup and Building Python guide, and `pyenv`'s [Common build problems](https://github.com/pyenv/pyenv/wiki/Common-build-problems) one should set `CPPFLAGS` and `LDFLAGS` like this:
> 
> ```sh
> CPPFLAGS="-I/opt/local/include" LDFLAGS="-L/opt/local/lib" pyenv install 3.10.8
> Downloading openssl-1.1.1o.tar.gz...
> -> https://www.openssl.org/source/openssl-1.1.1o.tar.gz
> Installing openssl-1.1.1o...
> Installed openssl-1.1.1o to /Users/thecesrom/.pyenv/versions/3.10.8
> 
> Downloading readline-8.1.tar.gz...
> -> https://ftpmirror.gnu.org/readline/readline-8.1.tar.gz
> Installing readline-8.1...
> Installed readline-8.1 to /Users/thecesrom/.pyenv/versions/3.10.8
> 
> Downloading Python-3.10.8.tar.xz...
> -> https://www.python.org/ftp/python/3.10.8/Python-3.10.8.tar.xz
> Installing Python-3.10.8...
> python-build: use zlib from xcode sdk
> Installed Python-3.10.8 to /Users/thecesrom/.pyenv/versions/3.10.8
> ```
> 
> Also, `pyenv` uses `openssl` and `readline` from Homebrew if present, something I am not seeing here.
> 
> So my question is.
> 
> How were you (@gleeblezoid) able to install a Python version with `pyenv` without setting those flags?
> 
> Thanks

I suspect this is one of those "worked on my machine" problems. I've recently wiped one of my devices so I'll try this port in isolation - I'd bet that said flags/prereqs were set from another install of something. It's possible to test this using a local port index fwiw (if you're also just interested in how to test a port from a branch off of macports-ports).

File: sysutils/pyenv/Portfile
Comment (gleeblezoid) [inline sysutils/pyenv/Portfile:29]: Prereqs included here @thecesrom (this section lists them as dependencies).

Comment (gleeblezoid) [reply sysutils/pyenv/Portfile:29]: I think the CPP and LD flags are also set with default values by MacPorts itself (again, remembering back based on when I originally pieced this together from the pyenv setup guides).

Comment (thecesrom) [reply sysutils/pyenv/Portfile:29]: I'll go ahead and create a local repo and share feedback. Thanks, @gleeblezoid!

Comment (gleeblezoid) [reply sysutils/pyenv/Portfile:29]: Yup, the defaults are listed in https://guide.macports.org/#reference.phases.configure:

- `LDFLAGS` `Default: -L${prefix}/lib -Wl,-headerpad_max_install_names`
- `CPPFLAGS` `Default: -I${prefix}/include`

Comment (gleeblezoid) [reply sysutils/pyenv/Portfile:29]: I'm still happy to re-test the install and setup on a wiped machine with no other ports but hopefully the above helps with the questions re prereq installs and the flags setup.

Comment (gleeblezoid) [reply sysutils/pyenv/Portfile:29]: I recall some of this being pretty fiddly way back when, so very grateful for any input you've got @thecesrom

**Comment by gleeblezoid** [comment]
Moving back to draft to avoid bugging contributors.

**Comment by thecesrom** [comment]
I created the local repo, and installed `pyenv` after making a few changes to your latest Portfile:

Changes:

```
github.setup            pyenv pyenv 2.3.5 v
...
categories              python
...
github.tarball_from     archive
checksums               rmd160 1f115021f728bc7e8be10f8475bd16af3d74f630 \
                        sha256 075e530afe352d112a23d46ba1a3358090bb9e31e47b5caa4ab1a1f32df0cfb0 \
                        size   716231
```

Since `pyenv` takes inspiration from `rbenv` I changed the category from `sysutils` to `python`, and updated to the latest version `2.3.5`.

## Installation

```sh
$ sudo port install pyenv
Password:
--->  Computing dependencies for pyenv
The following dependencies will be installed:
 autoconf
 m4
Continue? [Y/n]: y
--->  Fetching archive for m4
--->  Attempting to fetch m4-1.4.19_1.darwin_19.x86_64.tbz2 from https://packages.macports.org/m4
--->  Attempting to fetch m4-1.4.19_1.darwin_19.x86_64.tbz2.rmd160 from https://packages.macports.org/m4
--->  Installing m4 @1.4.19_1
--->  Activating m4 @1.4.19_1
--->  Cleaning m4
--->  Fetching archive for autoconf
--->  Attempting to fetch autoconf-2.71_1.darwin_19.noarch.tbz2 from https://packages.macports.org/autoconf
--->  Attempting to fetch autoconf-2.71_1.darwin_19.noarch.tbz2.rmd160 from https://packages.macports.org/autoconf
--->  Installing autoconf @2.71_1
--->  Activating autoconf @2.71_1
--->  Cleaning autoconf
--->  Fetching distfiles for pyenv
--->  Attempting to fetch pyenv-2.3.5.tar.gz from https://distfiles.macports.org/pyenv
--->  Attempting to fetch pyenv-2.3.5.tar.gz from https://ywg.ca.distfiles.macports.org/mirror/macports/distfiles/pyenv
--->  Attempting to fetch pyenv-2.3.5.tar.gz from https://github.com/pyenv/pyenv/archive/v2.3.5
--->  Verifying checksums for pyenv
--->  Extracting pyenv
--->  Configuring pyenv
--->  Building pyenv
--->  Staging pyenv into destroot
--->  Installing pyenv @2.3.5_0
--->  Activating pyenv @2.3.5_0
--->  Cleaning pyenv
--->  Scanning binaries for linking errors
--->  No broken files found.
--->  No broken ports found.
--->  Some of the ports you installed have notes:
  pyenv has the following notes:
    To configure your environment, add the following to your shell config file:
      export PYENV_ROOT='/opt/local/bin/pyenv'
      eval $(pyenv init -)
    You may need to add pyenv to your path.
```

I did run into an issue after I updated my shell config saying that`pyenv` could not be found as `PYENV_ROOT`, in my case, was `/opt/local/bin/pyenv-2.3.5` instead of `/opt/local/bin/pyenv`. After correcting that, `pyenv` requested the following settings on my `.zprofile`, where `pyenv` via `brew` was happy with it being on my `.zshrc`.

```sh
```

But still I did run into the same problem when installing the latest Python 3.10:

```sh
$ pyenv install 3.10.7
Downloading openssl-1.1.1o.tar.gz...
-> https://www.openssl.org/source/openssl-1.1.1o.tar.gz
Installing openssl-1.1.1o...
Installed openssl-1.1.1o to /Users/thecesrom/.pyenv/versions/3.10.7

Downloading readline-8.1.tar.gz...
-> https://ftpmirror.gnu.org/readline/readline-8.1.tar.gz
Installing readline-8.1...
Installed readline-8.1 to /Users/thecesrom/.pyenv/versions/3.10.7

Downloading Python-3.10.7.tar.xz...
-> https://www.python.org/ftp/python/3.10.7/Python-3.10.7.tar.xz
Installing Python-3.10.7...
python-build: use zlib from xcode sdk
WARNING: The Python lzma extension was not compiled. Missing the lzma lib?
Installed Python-3.10.7 to /Users/thecesrom/.pyenv/versions/3.10.7
```

So setting `CPPFLAGS` and `LDFLAGS` is a must, apparently.

And installing Python 2.7.18 I got the following output:

```sh
pyenv install 2.7.18
Error: No available formula with the name "openssl@1.0". Did you mean openssl@1.1 or openssl@3?
Downloading openssl-1.0.2q.tar.gz...
-> https://www.openssl.org/source/old/1.0.2/openssl-1.0.2q.tar.gz
Installing openssl-1.0.2q...
Installed openssl-1.0.2q to /Users/thecesrom/.pyenv/versions/2.7.18

Downloading readline-8.0.tar.gz...
-> https://ftpmirror.gnu.org/readline/readline-8.0.tar.gz
Installing readline-8.0...
Installed readline-8.0 to /Users/thecesrom/.pyenv/versions/2.7.18

Downloading Python-2.7.18.tar.xz...
-> https://www.python.org/ftp/python/2.7.18/Python-2.7.18.tar.xz
Installing Python-2.7.18...
patching file configure
patching file configure.ac
patching file setup.py
patching file Mac/Tools/pythonw.c
patching file setup.py
patching file Doc/library/ctypes.rst
patching file Lib/test/test_str.py
patching file Lib/test/test_unicode.py
patching file Modules/_ctypes/_ctypes.c
patching file Modules/_ctypes/callproc.c
patching file Modules/_ctypes/ctypes.h
patching file Modules/_ctypes/callproc.c
patching file setup.py
patching file Mac/Modules/qt/setup.py
patching file setup.py
python-build: use zlib from xcode sdk
Installed Python-2.7.18 to /Users/thecesrom/.pyenv/versions/2.7.18
```

Perhaps in that case I could only pass `openssl11` and `readline` via `CPPFLAGS` and `LDFLAGS`.

Something that was not an issue with Homebrew, and this is mainly because `pyenv`'s [`python-build`](https://github.com/pyenv/pyenv/blob/master/plugins/python-build/bin/python-build) script heavily relies on Homebrew.

**Comment by gleeblezoid** [comment]
Thanks for the work on this @thecesrom! I'll see if I can force the flag setting via the portfile (at worst adding it to the notes to set in local env) and dig into that script to see if I can replicate the full set of dependencies (I may need to refer to brew files to step back further into that). I really appreciate you pointing at the build script as that seems a goldmine of "what do".

**Comment by gleeblezoid** [comment]
After poking this with a sharp stick for way too long I finally figured out what I was doing wrong. Because pyenv gets installed under the macports prefix directory you need to run it with sudo (as it's configured at the moment) - which means env vars from .zshrc don't all get factored in when run using sudo (which includes the `CPPFLAGS` and `LDFLAGS` stuf). I got so sucked into digging in Python I forgot to think about the OS context. 🤦🏾

**Comment by pmetzger** [comment]
@reneeotten Do you think this is mergeable?

**Comment by reneeotten** [comment]
> @reneeotten Do you think this is mergeable?

I don't know - it has been forever with no clear follow-up or path forward, so I haven't looked at it for a while. I am not sure that installing another package to manage Python package by MacPorts is necessarily the greatest idea, but then again that's just my opinion.

**Comment by gleeblezoid** [comment]
In case it's useful to have the context - I've got this working as far as I think I can without also fixing the current issues within pyenv which essentially tie it to homebrew (which I was not aware of until digging a lot deeper into pyenv itself).

Given that doesn't look like it will be fixed any time soon (I don't really have the bandwidth to do it and it's been open a while) I suspect the correct answer is to close this off without merging (or whatever the preferred non-merge path is).

Much as the automated version swapping of pyenv is useful, it feels like it's in a bad place to be a port.

**Comment by pmetzger** [comment]
So should I close this for now?

**Comment by gleeblezoid** [comment]
> So should I close this for now?

I would be happy with that fwiw, if I do manage to come back to this via the route of a pyenv fix then I can always pick it up again. 

I'll close it out given I'm here right now and it feels like this is the way forward.

**Pull request closed without merging**
